docs/source/chaincode4noah.md

```diff
@@ -246,62 +244,61 @@ You can upgrade a chaincode using the same Fabric lifecycle process as you used
+without having to repackage your chaincode binaries. Channel members simply
+need to approve a definition with the new policy. The new definition needs to
+increment the **sequence** variable in the definition by one.
 
-    ![Approve a new chaincode definition](lifecycle/Lifecycle-upgrade-approve.png)
+  ![Approve a new chaincode definition](lifecycle/Lifecycle-upgrade-approve.png)
 
-  *Organization administrators from Org1 and Org2 approve the new chaincode
-  definition for their respective organizations. The new definition references
-  the new packageID and changes the chaincode version. Since this is the first
-  update of the chaincode, the sequence is incremented from one to two.*
+*Organization administrators from Org1 and Org2 approve the new chaincode
+definition for their respective organizations. The new definition references
+the new packageID and changes the chaincode version. Since this is the first
+update of the chaincode, the sequence is incremented from one to two.*
 
-4. **Commit the definition to the channel:** When a sufficient number of channel
-  members have approved the new chaincode definition, one organization can
-  commit the new definition to upgrade the chaincode definition to the channel.
-  There is no separate upgrade command as part of the lifecycle process.
+**Step four: Commit the definition to the channel:** When a sufficient number of
+channel members have approved the new chaincode definition, one organization can
+commit the new definition to upgrade the chaincode definition to the channel.
+There is no separate upgrade command as part of the lifecycle process.
 
-    ![Commit the new definition to the channel](lifecycle/Lifecycle-upgrade-commit.png)
+  ![Commit the new definition to the channel](lifecycle/Lifecycle-upgrade-commit.png)
 
-  *An organization administrator from Org1 or Org2 commits the new chaincode
-   definition to the channel. The chaincode containers are still running the old
-   chaincode.*
+*An organization administrator from Org1 or Org2 commits the new chaincode
+definition to the channel.*
 
-5. **Upgrade the chaincode container:** If you updated the chaincode definition
-  without upgrading the chaincode package, you do not need to upgrade the
-  chaincode container. If you did upgrade the chaincode binaries, a new invoke
-  will upgrade the chaincode container. If you requested the execution of the
-  ``Init`` function in the chaincode definition, you need to upgrade the
-  chaincode container by invoking the ``Init`` function again after the new
-  definition is successfully committed.
+After you commit the chaincode definition, a new chaincode container will
+launch with the code from the upgraded chaincode binaries. If you requested the
+execution of the ``Init`` function in the chaincode definition, you need to
+initialize the upgraded chaincode by invoking the ``Init`` function again after
+the new definition is successfully committed. If you updated the chaincode
```

Comment:
It's whether the version of the definition changed, not whether the code package changed.  Although typically, the version changing would imply the package changing as well, since each org decides which package to run, the only common value/indicator we have is the 'version' field of the definition.

Reply:
I updates this section, but also added some more detail to the step above.
`You also need to change the chaincode version, which is used by the lifecycle process to track if the chaincode binaries have been upgraded.`